### PR TITLE
Installation Markdowns

### PR DESCRIPTION
**Pull request by arunthurai**
## Proposed changes

Describe the changes implemented in this pull request. If the changes fix a bug or resolves a feature request, be sure to link the issue. Please explain the issue and how the changes address the issue.

## Types of changes

What types of changes does your code introduce? Put an `x` in the boxes that apply

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionalitiy)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [ ] Changes have been tested to ensure that fix is effective or that a feature works.
- [ ] Changes pass the unit tests
- [ ] Code has been run through the `poe quality` task
- [ ] I have included necessary documentation or comments (as necessary)
- [ ] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

### CODE REVIEW

File: docs/contributing/contributing.md
Comment (tkkuehn) [inline docs/contributing/contributing.md:7]: I would soften this a fair bit because if we deploy to PyPI people can just `pip install` this.

File: docs/contributing/contributing.md
Comment (tkkuehn) [inline docs/contributing/contributing.md:75]: This bit doesn't exist in labelmerge (right now). Should either implement it or get rid of this paragraph.

File: docs/getting_started/docker.md
Comment (tkkuehn) [inline docs/getting_started/docker.md:1]: This is maybe still useful but if we can pip install labelmerge that's probably even easier (at least in some cases)

File: docs/getting_started/docker.md
Comment (tkkuehn) [inline docs/getting_started/docker.md:12]: Missing closing backtick after `cd

File: docs/getting_started/singularity.md
Comment (tkkuehn) [inline docs/getting_started/singularity.md:56]: Just a note that we should actually do this before merging this PR

File: docs/getting_started/singularity.md
Comment (tkkuehn) [inline docs/getting_started/singularity.md:8]: Can soften these requirements a lot (probably just get rid of this list item)

File: docs/getting_started/singularity.md
Comment (tkkuehn) [inline docs/getting_started/singularity.md:12]: Probably don't need 8 cores and 64GB memory for an example Labelmerge

File: docs/getting_started/singularity.md
Comment (tkkuehn) [inline docs/getting_started/singularity.md:61]: This CLI invocation is for SCATTR, so update to Labelmerge

File: docs/getting_started/singularity.md
Comment (tkkuehn) [inline docs/getting_started/singularity.md:91]: SCATTR -> Labelmerge

File: docs/usage/cli.md
Comment (tkkuehn) [inline docs/usage/cli.md:8]: `neede` -> needed

File: docs/getting_started/installation.md
Comment (tkkuehn) [inline docs/getting_started/installation.md:11]: I don't think this is true for Labelmerge

File: docs/getting_started/installation.md
Comment (tkkuehn) [inline docs/getting_started/installation.md:7]: Or pip? We should decide if we want to release this via PyPI

File: docs/getting_started/docker.md
Comment (kaitj) [reply docs/getting_started/docker.md:1]: I agree though I would leave the container stuff in to give people different options

File: docs/usage/cli.md
Comment (kaitj) [reply docs/usage/cli.md:22]: It shouldn’t be for labelmerge. It was in scattr because there are rules that write to existing derivative folders as a result of using freesurfer

File: docs/getting_started/docker.md
Comment (kaitj) [inline docs/getting_started/docker.md:28]: Hte -> the

File: docs/getting_started/docker.md
Comment (kaitj) [inline docs/getting_started/docker.md:72]: Scattr related documentation, can be deleted

File: docs/getting_started/installation.md
Comment (kaitj) [inline docs/getting_started/installation.md:27]: Macines -> machines

File: docs/getting_started/singularity.md
Comment (kaitj) [reply docs/getting_started/singularity.md:12]: I don’t think we need more than 8gb maybe 16g for labelmerge

File: docs/workflow/simple_workflow.png
Comment (kaitj) [inline docs/workflow/simple_workflow.png:1]: I would modify this image for the repository to just show what labelmerge does. It’s not clear what the “original application” is referring to here.

**Review by kaitj — CHANGES_REQUESTED** [review]

File: docs/contributing/contributing.md
Comment (kaitj) [inline docs/contributing/contributing.md:11]: Would recommend using a version-tagged container if possible

File: docs/contributing/contributing.md
Comment (kaitj) [inline docs/contributing/contributing.md:27]: This feels like this should be a separate part of the doc on how to run labelmerge rather than contribution.

File: docs/contributing/contributing.md
Comment (kaitj) [inline docs/contributing/contributing.md:44]: labelmerge -> Labelmerge

File: docs/getting_started/docker.md
Comment (kaitj) [inline docs/getting_started/docker.md:40]: This is referencing scattr

File: docs/getting_started/docker.md
Comment (kaitj) [inline docs/getting_started/docker.md:50]: Spelling of labelmerge

File: docs/getting_started/singularity.md
Comment (kaitj) [inline docs/getting_started/singularity.md:39]: scattr reference

File: docs/getting_started/singularity.md
Comment (kaitj) [inline docs/getting_started/singularity.md:46]: scattr reference

File: docs/getting_started/singularity.md
Comment (kaitj) [inline docs/getting_started/singularity.md:53]: scattr reference

File: docs/getting_started/singularity.md
Comment (kaitj) [inline docs/getting_started/singularity.md:63]: Labelmerge spelling

File: docs/usage/cli.md
Comment (kaitj) [inline docs/usage/cli.md:27]: scattr reference

File: docs/usage/faq.md
Comment (kaitj) [inline docs/usage/faq.md:13]: Points to scattr docs

File: docs/workflow/workflow.md
Comment (kaitj) [inline docs/workflow/workflow.md:7]: "An simplified" -> "A simplified"

**Review by kaitj — CHANGES_REQUESTED** [review]
There are still a few references to scattr, as well as misspelling of labelmerge. Would also try to be consistent with capitalization (labelmerge vs Labelmerge), which is used interchangeably throughout.

File: docs/workflow/workflow.md
Comment (kaitj) [inline docs/workflow/workflow.md:30]: "pre-appended" -> "prepend"

File: docs/workflow/workflow.md
Comment (kaitj) [inline docs/workflow/workflow.md:28]: Is there a way to reword this to make it more clear - wording slightly confusing.

**Review by kaitj — CHANGES_REQUESTED** [review]

File: docs/workflow/workflow.md
Comment (kaitj) [inline docs/workflow/workflow.md:24]: If you can, would provide a link to the BEP that has the specs. 

Also "Brain Masks" -> "brain masks"

File: docs/workflow/workflow.md
Comment (kaitj) [inline docs/workflow/workflow.md:26]: This should be arranged accordingly since you mention its done "in order" and "overwrites". I believe its something like base -> overlay -> exceptions. Or would rephrase to make more clear.

File: docs/getting_started/singularity.md
Comment (kaitj) [reply docs/getting_started/singularity.md:59]: fs-license is still mentioned here.

File: docs/contributing/contributing.md
Comment (kaitj) [inline docs/contributing/contributing.md:10]: I think we can omit the word "singularity" here.

File: docs/contributing/contributing.md
Comment (kaitj) [inline docs/contributing/contributing.md:25]: Also need to capitalize "labelmerge" I think for consistency.

File: docs/contributing/contributing.md
Comment (kaitj) [inline docs/contributing/contributing.md:49]: `flake8` is going to change to `ruff` in an upcoming PR. I can make the change to the docs once this is merged in, otherwise I wouldn't mind if you changed it here either.

File: docs/getting_started/docker.md
Comment (kaitj) [inline docs/getting_started/docker.md:3]: Hmm, not sure what formatting this was meant to be - maybe italicized? I don't think the way `**` is used here will actually result in any sort of formatting and will be rendered as is.

File: docs/getting_started/docker.md
Comment (kaitj) [inline docs/getting_started/docker.md:20]: Reword -> "can be then be run"

File: docs/getting_started/installation.md
Comment (kaitj) [inline docs/getting_started/installation.md:49]: I think the requirements should be updated to include python if are including pip install instructions. Would also include pros / cons similar to the others if possible.

File: docs/getting_started/installation.md
Comment (kaitj) [inline docs/getting_started/installation.md:10]: Should provide a link to the BIDS derivatives spec so that users can see an example.

**Review by kaitj — CHANGES_REQUESTED** [review]

File: docs/usage/cli.md
Comment (kaitj) [inline docs/usage/cli.md:3]: labelmerge -> Labelmerge

File: docs/usage/cli.md
Comment (kaitj) [inline docs/usage/cli.md:6]: labelmerge -> Labelmerge

File: docs/usage/cli.md
Comment (kaitj) [inline docs/usage/cli.md:20]: labelmerge -> Labelmerge

File: docs/requirements.txt
Comment (arunthurai) [inline docs/requirements.txt:1]: include python3.7 here as a requirement? and would we also need pip then

Comment (kaitj) [reply docs/requirements.txt:1]: This `requirements.txt` is solely for the documentation and isn't actually part of the package (see `pyproject.toml` for those dependencies.

File: docs/getting_started/singularity.md
Comment (tkkuehn) [reply docs/getting_started/singularity.md:8]: This comment still stands -- I don't think people need anything like 16GB memory to run this

File: docs/usage/cli.md
Comment (tkkuehn) [inline docs/usage/cli.md:1]: The autogenerated arguments as it renders on readthedocs don't seem to include the arguments defined in the config. Those are really important, and I think it's worth going through them explicitly anyway. (i.e. `--overlay_bids_dir`, `--base_desc`, `--overlay_desc`, and the exceptions and drops args)

File: docs/usage/cli.md
Comment (tkkuehn) [inline docs/usage/cli.md:7]: I don't even really think the last bit is true -- you're almost always going to need either `--overlay_bids dir` or `--base_desc` and `--overlay_desc`

**Review by tkkuehn — CHANGES_REQUESTED** [review]
Almost there, just a few suggestions that I think will help these docs be maximally useful

File: docs/usage/cli.md
Comment (arunthurai) [reply docs/usage/cli.md:1]: Otherwise can manually write those for here

File: docs/usage/cli.md
Comment (arunthurai) [reply docs/usage/cli.md:7]: changed to include thse

File: docs/getting_started/installation.md
Comment (arunthurai) [reply docs/getting_started/installation.md:8]: added to cli section

File: docs/usage/cli.md
Comment (kaitj) [reply docs/usage/cli.md:1]: Yeah, you will need to add the `add_dynamic_args` in if it isn't already there.

**Review by tkkuehn — COMMENTED** [review]
Need to run `black` on this for the docs to build, but the changes are looking good

File: labelmerge/run.py
Comment (tkkuehn) [inline labelmerge/run.py:11]: Need to add `from snakebids.cli import add_dynamic_args` above with the other imports for this to work

**Comment by arunthurai** [comment]
Checks look like they pass

**Review by kaitj — APPROVED** [review]

**Pull request merged**
